dbms/src/Interpreters/Join.cpp

```diff
@@ -1731,27 +1731,38 @@ struct CrossJoinAdder;
 template <ASTTableJoin::Strictness STRICTNESS>
 struct CrossJoinAdder<ASTTableJoin::Kind::Cross, STRICTNESS>
 {
-    static void addFound(MutableColumns & dst_columns, size_t num_existing_columns, ColumnRawPtrs & src_left_columns, size_t num_columns_to_add, size_t start_offset, size_t i, const BlocksList & blocks, IColumn::Filter * is_row_matched, IColumn::Offset & current_offset, IColumn::Offsets * expanded_row_size_after_join)
+    static size_t calTotalRightRows(const BlocksList & blocks)
     {
-        size_t expanded_row_size = 0;
+        size_t total_rows = 0;
         for (const Block & block_right : blocks)
         {
             size_t rows_right = block_right.rows();
             if constexpr (STRICTNESS == ASTTableJoin::Strictness::Any)
```

Comment:
I think we should update `total_rows` as `total_rows = std::min(total_rows, 1)` after total_rows calculation is done? Otherwise, for `Any` strictniss, the total_rows is the total non-empty block size.

Reply:
You're right, I'll fixed it.

Reply:
> I think we should update `total_rows` as `total_rows = std::min(total_rows, 1)` after total_rows calculation is done? Otherwise, for `Any` strictniss, the total_rows is the total non-empty block size.

Done